### ISSUE TITLE
Resource aws_sns_topic_subscription, new argument redrive_policy

### ISSUE THREAD

**Issue by Cayce** [hashicorp/terraform-provider-aws#10931]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS implemented a new feature for SNS subscription: support for Dead-Letter Queues
https://aws.amazon.com/about-aws/whats-new/2019/11/amazon-sns-adds-support-for-dead-letter-queues-dlq/
The request is to implement it in resource _aws_sns_topic_subscription_, please add _redrive_policy_ as argument of this resource.

### New or Affected Resource

* aws_sns_topic_subscription

### Potential Terraform Configuration

```hcl
resource "aws_sns_topic_subscription" "lamda_function_sns_subscription" {
  topic_arn       = "arn:aws:sns:us-east-1:***:TOPIC_NAME"
  protocol        = "lambda"
  endpoint        = "arn:aws:lambda:us-east-1:***:function:function-name"
  redrive_policy = <<EOF
    {
      "deadLetterTargetArn": "arn:aws:sqs:us-east-1:***:queue-dlq"
    }
  EOF
}
```

**Comment by Ricool06** [comment]
PR #11770 can fix this, tested it manually, but never written Go before so unsure of how to unit and ACC test at the moment.

**Comment by Ricool06** [comment]
Okay so I'm in the middle of writing tests for the PR, and it seems you cannot just remove the RedrivePolicy attribute from a subscription. Once set, it requires a redrive policy with a valid arn for the deadLetterTargetArn attribute. I'm going to push the commits so far without the attribute removal test to get some visibility on it.

**Comment by Ricool06** [comment]
You can disable the redrive policy by making sure AttributeValue for RedrivePolicy is not set in the request. Just figuring out how to do that for the request to AWS.

**Comment by Ricool06** [comment]
Fixed it, sorry for spam, GitHub mobile won't let me edit. #11770 is ready with ACC tests passing.

**Comment by kabo** [comment]
Any update on this? Sure could use this right about now :D

**Comment by Ricool06** [comment]
> Any update on this? Sure could use this right about now :D

I've attached a pull request, ready to go, but I'm yet to get any of the terraform peeps to review. If you could show/tell me how to get their attention a bit, I would appreciate it. Happy to progress this 👍

**Comment by gageorsburn** [comment]
Also in need of this feature.

**Comment by jdrowne** [comment]
Reiterating we have a real life need for this feature.

**Comment by amalbuquerque** [comment]
Also depending on this. Do we have an ETA for @Ricool06 PR to be merged?

Thanks in advance!

**Comment by Ricool06** [comment]
@amalbuquerque I am waiting on a review from the terraform people, until then, I am unable to merge. ☹️

The PR has been ready for review for months.

**Comment by isikdos** [comment]
We have a real life need for this feature, Adding my support to increase visibility for this item.

**Comment by gustavoamigo** [comment]
Does anyone have a workaround to this? Maybe using CloudFormation or something?

**Comment by Ricool06** [comment]
@gustavoamigo the workaround until my PR gets merged is to use a null resource + aws cli

```terraform
resource "null_resource" "redrive_policy" {
  provisioner "local-exec" {
    command = "aws sns <put aws cli command here>"
  }
}
```

Make sure that it depends on your sns topic/subscription resource so the cli command runs only after the necessary resources are created.

**Comment by ssteuteville** [comment]
Also need this.

**Comment by Ricool06** [comment]
> Also need this.

If you can get @bflad or any other terraform maintainer to review and merge this PR (#11770): https://github.com/terraform-providers/terraform-provider-aws/pull/11770 then we can have it. 👍

**Comment by rbcrwd** [comment]
'twould be nice to have this. the workaround a few comments up is helpful, but the need is real and it looks like we have a PR ready for this.

**Comment by davebrown211** [comment]
also need this

**Comment by Ricool06** [comment]
Okay, so I think the PR bot may have bugged out and that's why it's not getting any reviews. I'm going to open a new PR and hopefully it gets reviewed and merged.
Apologies if this goes against the rules a bit.

**Comment by Ricool06** [comment]
#14800 hopefully resolves the issue with the PR bot.

**Comment by breathingdust** [comment]
Hi all! :wave: Just wanted to direct you to our public [roadmap](https://github.com/hashicorp/terraform-provider-aws/blob/master/ROADMAP.md) for this quarter (Nov-Jan) in which this item has been mentioned.

Due to the significant community interest in support for this feature, we will be looking at merging existing contributions soon. 

We appreciate all the contributions and feedback thus far.

Look out for support in the provider soon!

**Comment by Ricool06** [comment]
> Hi all!  Just wanted to direct you to our public [roadmap](https://github.com/hashicorp/terraform-provider-aws/blob/master/ROADMAP.md) for this quarter (Nov-Jan) in which this item has been mentioned.
> 
> Due to the significant community interest in support for this feature, we will be looking at merging existing contributions soon.
> 
> We appreciate all the contributions and feedback thus far.
> 
> Look out for support in the provider soon!

Hey, it seems the roadmap references the wrong issue number. It is supposed to be #10931 but it says #1093.

**Comment by jacoor** [comment]
#1093: Resource aws_sns_topic_subscription, new argument redrive_policy is mentioned in Worflow improvements. 
Looking forward to it.

**Comment by jacoor** [comment]
yup, as @Ricool06 mentioned, the number seems wrong on the roadmap. @breathingdust can you look into it?

**Comment by breathingdust** [comment]
Fixed, thanks for pointing that out @Ricool06 / @jacoor 🎉

**Comment by jacoor** [comment]
phew :-)

**Comment by abhi-dwivedi** [comment]
Any Updates on this ??

**Comment by Ricool06** [comment]
@abhi-dwivedi I'm resigned to the belief we're probably not gonna get anywhere with this. The PR I made is a year out of date and as much as I would like to update it, I get the feeling it may get forgotten again.

**Comment by yumingqiao** [comment]
It would be nice if this can be supported
